streamingData not found

I am getting trying to get video data using the doc but is throwing me a BadPageException

Videos tested: 
e3l1piy1pn8
https://www.youtube.com/watch?v=GmG4X9PGOXs
https://www.youtube.com/watch?v=5UW3oWn60D0

Well i tested more videos and always i get the same error
Gist: https://gist.github.com/SrRapero720/ecf46d27d76383ca3ecf098c9af8fe74

Aditional Context:
The package isn't the original, because i shadow the package in the jar. because i am working on a minecraft mod doing rendering stuff. and other mods implements this dependency but with old versions and exporting the same package cause crashes.

code used:
```java
    @Nullable
    public static String getYoutubeRawURL(String videoUrl) throws MalformedURLException {
        // I tested with both cases, using videoID and videoUrl. nothing works
        var videoID = isAValidYoutubeURL(videoUrl) ? getVideoID(new URL(videoUrl)).get("v").get(0) : DEFAULT_URL;
//        var videoID = videoUrl;
        WaterFrames.LOGGER.info(videoID);
        synchronized (CACHE) {
            var hit = CACHE.get(videoID);
            if (hit != null && ((hit.url != null && System.currentTimeMillis() - hit.time < 1000 * 60 * 10) || (hit.url == null && System.currentTimeMillis() - hit.time < 1000 * 60))) return hit.url();
            var mediaInfo = youtubeDownloader.getVideoInfo(new RequestVideoInfo(videoID).callback(new YoutubeCallback<VideoInfo>() {
                @Override
                public void onFinished(VideoInfo data) {
                    String url = data.bestVideoWithAudioFormat().url();
                    WaterFrames.LOGGER.info(url);
                }

                @Override
                public void onError(Throwable throwable) {
                    WaterFrames.LOGGER.error("failed", throwable);
                }
            }));
            var video = mediaInfo.data();

            if (mediaInfo.error() != null) throw new RuntimeException(mediaInfo.error());


            String url = video.bestVideoWithAudioFormat().url();
            hit = new URLCacheEntry(System.currentTimeMillis(), url);
            CACHE.put(videoID, hit);
            return hit.url;
        }
    }
```

sadly I can't modify the code directly, I just did some JarInJar to make it work (and rename the package using shadowjar)

In any case, using a tool called mixin that modifies the bytecode at runtime I was able to eliminate that line of code and now the error is different.

```log
[17:42:09] [WATERCoRE-45-65181655849003/ERROR] [waterframes/]: Failed waterframes to load:
17:42:09.056
game
java.lang.RuntimeException: me.srrapero720.github.kiulian.downloader.YoutubeException$BadPageException: streamingData and videoDetails not found
17:42:09.056
game
at me.srrapero720.waterframes.ytdl.YoutubeUtil.getYoutubeRawURL(YoutubeUtil.java:64) ~[WATERFrAMES-1.1.0-ALPHA8.jar%2366!/:1.1.0-ALPHA8] {re:classloading}
17:42:09.056
game
at me.srrapero720.waterframes.display.MediaDisplay.lambda$new$1(MediaDisplay.java:124) ~[WATERFrAMES-1.1.0-ALPHA8.jar%2366!/:1.1.0-ALPHA8] {re:classloading}
17:42:09.057
game
at me.srrapero720.watercore.api.thread.ThreadUtil.lambda$threadTry$1(ThreadUtil.java:21) ~[WATERCoRE-1.4.0-BETA20.jar%2365!/:1.4.0-BETA20] {re:classloading}
17:42:09.057
game
at me.srrapero720.watercore.api.thread.ThreadUtil.lambda$threadTryArgument$3(ThreadUtil.java:34) ~[WATERCoRE-1.4.0-BETA20.jar%2365!/:1.4.0-BETA20] {re:classloading}
17:42:09.057
game
at java.lang.Thread.run(Thread.java:833) [?:?] {}
17:42:09.057
game
Caused by: me.srrapero720.github.kiulian.downloader.YoutubeException$BadPageException: streamingData and videoDetails not found
17:42:09.057
game
at me.srrapero720.github.kiulian.downloader.parser.ParserImpl.parseVideoWeb(ParserImpl.java:153) ~[WATERFrAMES-1.1.0-ALPHA8.jar%2366!/:1.1.0-ALPHA8] {re:mixin,re:classloading,pl:mixin:APP:waterframes.mixin.json:YoutubeDownloaderMixin,pl:mixin:A}
17:42:09.057
game
at me.srrapero720.github.kiulian.downloader.parser.ParserImpl.parseVideo(ParserImpl.java:65) ~[WATERFrAMES-1.1.0-ALPHA8.jar%2366!/:1.1.0-ALPHA8] {re:mixin,re:classloading,pl:mixin:APP:waterframes.mixin.json:YoutubeDownloaderMixin,pl:mixin:A}
17:42:09.057
game
at me.srrapero720.github.kiulian.downloader.parser.ParserImpl.parseVideo(ParserImpl.java:53) ~[WATERFrAMES-1.1.0-ALPHA8.jar%2366!/:1.1.0-ALPHA8] {re:mixin,re:classloading,pl:mixin:APP:waterframes.mixin.json:YoutubeDownloaderMixin,pl:mixin:A}
17:42:09.057
game
at me.srrapero720.github.kiulian.downloader.YoutubeDownloader.getVideoInfo(YoutubeDownloader.java:56) ~[WATERFrAMES-1.1.0-ALPHA8.jar%2366!/:1.1.0-ALPHA8] {re:classloading}
17:42:09.057
game
at me.srrapero720.waterframes.ytdl.YoutubeUtil.getYoutubeRawURL(YoutubeUtil.java:50) ~[WATERFrAMES-1.1.0-ALPHA8.jar%2366!/:1.1.0-ALPHA8] {re:classloading}
17:42:09.057
game
... 4 more
```

the only change i do is redirect to send a null value and skip to parseVideoWeb
![image](https://user-images.githubusercontent.com/29090346/233215797-f8afd67d-3018-4eaf-8f41-9a6f82ed6a9f.png)

--- a/src/main/java/com/github/kiulian/downloader/parser/ParserImpl.java
+++ b/src/main/java/com/github/kiulian/downloader/parser/ParserImpl.java
@@ -80,8 +80,12 @@ public class ParserImpl implements Parser {
                 "    \"client\": {" +
                 "      \"hl\": \"en\"," +
                 "      \"gl\": \"US\"," +
-                "      \"clientName\": \"ANDROID\"," +
-                "      \"clientVersion\": \"16.02\"" +
+                "      \"clientName\": \"TV_UNPLUGGED_ANDROID\"," +
+                "      \"clientVersion\": \"1.37\"," +
+                "      \"androidSdkVersion\": 31," +
+                "      \"osName\": \"Android\"," +
+                "      \"osVersion\": \"12\"," +
+                "      \"platform\": \"MOBILE\"" +
                 "    }" +
                 "  }" +
                 "}";

--- a/src/main/java/com/github/kiulian/downloader/parser/ParserImpl.java
+++ b/src/main/java/com/github/kiulian/downloader/parser/ParserImpl.java
@@ -80,12 +80,9 @@ public class ParserImpl implements Parser {
                 "    \"client\": {" +
                 "      \"hl\": \"en\"," +
                 "      \"gl\": \"US\"," +
-                "      \"clientName\": \"TV_UNPLUGGED_ANDROID\"," +
-                "      \"clientVersion\": \"1.37\"," +
-                "      \"androidSdkVersion\": 31," +
-                "      \"osName\": \"Android\"," +
-                "      \"osVersion\": \"12\"," +
-                "      \"platform\": \"MOBILE\"" +
+                "      \"clientName\": \"ANDROID_TESTSUITE\"," +
+                "      \"clientVersion\": \"1.9\"," +
+                "      \"androidSdkVersion\": 31" +
                 "    }" +
                 "  }" +
                 "}";

